### PR TITLE
feat: added BM extenxtion cartridge for Imgix

### PR DESCRIPTION
**Pull request by SVPToptal**

### CODE REVIEW

**Review by frederickfogerty — CHANGES_REQUESTED** [review]
`imagix` should be `imgix` - this is incorrect in the folder name and also in the `.project` file

File: cartridges/bm_app_imagix_base/.project [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<projectDescription>
+    <name>bm_app_imagix_base</name>
```
Comment (frederickfogerty) [inline cartridges/bm_app_imagix_base/.project:3]:
```suggestion
    <name>bm_app_imgix_base</name>
```

**Comment by luqven** [comment]
@SVPToptal could you please add a description of the change this PR encompasses?

I'm struggling just by reading the changes here to understand what committing this code would change for our Page Designer component.

I find it helpful when PRs that change behavior / UI in some way include a video. You can try out Vimeo's [extension](https://chrome.google.com/webstore/detail/vimeo-record-screen-webca/ejfmffkmeigkphomnpabpdabfddeadcb?hl=en) if you don't already have a good way of recording. If this change is easier to explain in a screen recording please feel free to do that instead.

**Review by luqven — CHANGES_REQUESTED** [review]
Hey @SVPToptal, a couple of comments.

1) Please make sure you have a clean commit history before asking for a review
2) Please make sure you use camel or snake case when naming files and folders to avoid spelling mistakes
3) Please fix the typo in `/cartridges/int_imgix_pd/cartridge/app_imgix_base_sfra.properties`
4) **Please let me know if the current directory structure is working on your end. If it's not, let me know if we need to change it as I suggest bellow.**

---
**First**
 I’ve cleaned up your commit history. Please make sure to familiarize yourself with GIT's interactive rebase to do this yourself in the future.

**Second**
I’ve fixed a spelling mistake for `imgrender` to `imagerenderer`. To avoid spelling mistakes like this, it's helpful to camel or snake case folder and file names. I’ve gone ahead and renamed the file and folder to `imageRenderer` for this reason.

**Third**
 In `/cartridges/int_imgix_pd/cartridge/app_imgix_base_sfra.properties`, `app_inix_base_sfra` was probably supposed to be `app_imgix_base_sfra`.

**Finally, and most importantly**
The current file structure looks like this:

```
├── cartridges
│   ├── app_imgix_base_sfra
│   │   └── cartridge
│   │       ├── app_imgix_base_sfra.properties
│   │       ├── experience
│   │       │   └── components
│   │       │       └── imgix
│   │       │           ├── testcomp.js
│   │       │           └── testcomp.json
│   │       └── templates
│   │           └── default
│   │               └── imgix
│   │                   └── image
│   │                       └── imgerender
│   │                           └── imagerender.isml
│   └── bm_app_imgix_base
│       └── cartridge
│           ├── experience
│           │   └── editors
│           │       └── custom
│           │           ├── tileCustomView.js
│           │           └── tileCustomView.json
│           └── static
│               └── default
│                   └── experience
│                       └── editors
│                           └── displayCustomView.js
```

Should it not instead look more like this?

```
├── cartridges
│   ├──app_imgix_base_sfra
│   │   └── cartridge
│   │       ├── app_imgix_base_sfra.properties
│   │       ├── experience
│   │       │   └── components
│   │       │       └── assets
│   │       │           ├── testcomp.js
│   │       │           └── testcomp.json
│   │       └── templates
│   │           └── default
│   │               └── experience
│   │                   └── components
│   │                       └── assets
│   │                           └── imageRender.isml
│   └──bm_imgix_pd
│      └── cartridge
│          ├── experience
│          │   └── editors
│          │       └── imgix
│          │           ├── tileCustomView.js
│          │           └── tileCustomView.json
│          └── static
│              └── default
│                  └── experience
│                      └── editors
│                          └── imgix
│                              └── displayCustomView.js
```

SFCC expects a very specific folder structure. If we don’t follow it, our component won’t work as expected. It seems to me like we need to change the file structure to better match what SFCC expects. 

Please let me know if you have any inclination as to whether or not that needs to be done.
----

To recap:

1) Please make sure you have a clean commit history before asking for a review
2) Please make sure you use camel or snake case when naming files and folders to avoid spelling mistakes
3) Please fix the typo in `/cartridges/int_imgix_pd/cartridge/app_imgix_base_sfra.properties`
4) **Please let me know if the current directory structure is working on your end. If it's not, let me know if we need to change it as I suggest above.**

File: cartridges/bm_app_imgix_base/cartridge/experience/editors/custom/tileCustomView.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+    "name": "IMGIX product view",
```
Comment (luqven) [inline cartridges/bm_app_imgix_base/cartridge/experience/editors/custom/tileCustomView.json:2]: FYI, imgix should never be capitalized.

```suggestion
    "name": "imgix product view",
```

File: cartridges/bm_app_imgix_base/cartridge/static/default/experience/editors/displayCustomView.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    // When a value was selected
+    listen('sfcc:value', value => { });
+    // When the editor must require the user to select something
+    listen('sfcc:required', value => { });
+    // When the editor is asked to disable its controls
+    listen('sfcc:disabled', value => {
+      if (rootEditorElement) {
+        rootEditorElement.querySelector('.btn-group').disabled = true;
+      }
+    });
```
Comment (luqven) [inline cartridges/bm_app_imgix_base/cartridge/static/default/experience/editors/displayCustomView.js:28]: Are lines 19 - 28 doing anything? If not we should not commit them.

```suggestion
```

File: cartridges/bm_app_imgix_base/cartridge/experience/editors/custom/tileCustomView.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    "description": "Returns a selector for the recommenders available on the instance. Allows product-to-product recommender",
```
Comment (luqven) [inline cartridges/bm_app_imgix_base/cartridge/experience/editors/custom/tileCustomView.json:3]: I'm not sure what this description means. Can you expand on this? What is a "selector for recommenders"?

File: cartridges/bm_app_imgix_base/cartridge/experience/editors/custom/tileCustomView.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+'use strict';
```
Comment (luqven) [inline cartridges/bm_app_imgix_base/cartridge/experience/editors/custom/tileCustomView.js:1]: I'm not sure what this file is doing. Can you explain what it does and why we need it? If it's not needed we shouldn't commit it.

**Review by luqven — CHANGES_REQUESTED** [review]
A couple of other questions

File: cartridges/bm_app_imgix_base/cartridge/static/default/experience/editors/displayCustomView.js (hunk 3 above)
Comment (SVPToptal) [reply cartridges/bm_app_imgix_base/cartridge/static/default/experience/editors/displayCustomView.js]: we will remove this future  if we not use this, now just added to test the component rendering flow for all

File: cartridges/bm_app_imgix_base/cartridge/experience/editors/custom/tileCustomView.js (hunk 5 above)
Comment (SVPToptal) [reply cartridges/bm_app_imgix_base/cartridge/experience/editors/custom/tileCustomView.js]: all SFCC script file require this line, then only server can read this file. it's SFCC backend related.

**Comment by SVPToptal** [comment]
As per the SFCC standard we use camelCase for Script and Controller files. for Folder and ISML  all should be on Small letter. i update the file accordingly.

**Review by luqven — APPROVED** [review]
LGTM 👍🏼 

I still think a discussion around the folder structure is necessary at some point in the future.

**Comment by SVPToptal** [comment]
> `imagix` should be `imgix` - this is incorrect in the folder name and also in the `.project` file

Done the change, please review

**Comment by luqven** [comment]
Apologies for all the force pushes here @SVPToptal, our branches got out of sync.
